src/coreclr/inc/winwrap.h

```diff
@@ -254,18 +254,32 @@ WszCreateProcess(
     LPPROCESS_INFORMATION lpProcessInformation
     );
 
-#if defined(HOST_X86) && defined(_MSC_VER)
+#if defined(_MSC_VER)
 
 //
-// Windows SDK does not use intrinsics on x86. Redefine the interlocked operations to use intrinsics.
+// Redefine the interlocked operations to intrinsics
+// to avoid missing InterlockedCompareExchange8 and Windows 8 requirement.
```

Comment:
What is the Windows 8 requirement that this is talking about?

Reply:
According to the documentation, [InterlockedExchange8](https://learn.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedexchange8) and [InterlockedCompareExchange16](https://learn.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedcompareexchange16) are exposed only on Windows 8+. My reasoning was that even though Windows 7 is not supported anymore, working this around here makes more sense than totally breaking the API there especially since we need to that already for the completely missing [InterlockedCompareExchange8](https://godbolt.org/z/9bWjvq91W).

Reply:
InterlockedExchange8 and InterlockedCompareExchange16 are always inlined as compiler intrinsics. The underlying OS does not matter. 

My guess is that nothing new is mentioned in the docs as supported on Win7, even when it is works there just fine.

I would keep this as simple as possible.

Reply:
It seems like `InterlockedExchange8` [is missing too on ARM64 MSVC](https://godbolt.org/z/v39Trfae6).

> InterlockedExchange8 and InterlockedCompareExchange16 are always inlined as compiler intrinsics. The underlying OS does not matter.

Since the docs say `This function is implemented using a compiler intrinsic where possible.`, I assumed the compiler might emit a call to a method on some platforms.
Even then, I don't think doing this selectively makes sense here, if some intrinsics require redefining to work properly, it makes more sense to consistently redefine all of them in such case.

Reply:
> It seems like InterlockedExchange8 [is missing too on ARM64 MSVC](https://godbolt.org/z/v39Trfae6).

The define for InterlockedExchange8 can stay then.

I see these defines as workarounds for Windows SDK bugs. Once these bugs are fixed, these workarounds can be deleted.

We just need the minimum required to work around the bugs in the current Windows SDK. We do not need to grow this into something complete or permanent.

Reply:
I have pushed a commit to keep make the workaround minimal and reference the issue.

Reply:
> I have pushed a commit to keep make the workaround minimal and reference the issue.

I think you missed `InterlockedExchange8` there.

Reply:
`InterlockedExchange8` is there and it works fine as far as I can tell.